test-e2e/byte-range.test.ts

```diff
@@ -1,20 +1,29 @@
-import { test, expect } from '@playwright/test'
-import { waitForServiceWorker } from './fixtures/wait-for-service-worker.js'
+import { testPathRouting as test, expect } from './fixtures/config-test-fixtures.js'
+import { doRangeRequest } from './fixtures/do-range-request.js'
 
 test.describe('byte-ranges', () => {
   test('should be able to get a single character', async ({ page }) => {
-    await page.goto('/', { waitUntil: 'networkidle' })
-    // wait for service worker to load
-    await waitForServiceWorker(page)
+    const { text, byteSize, statusCode } = await doRangeRequest({ page, range: 'bytes=1-2', path: '/ipfs/bafkqaddimvwgy3zao5xxe3debi' })
 
-    const partialText = await page.evaluate(async () => {
-      const response = await fetch('/ipfs/bafkqaddimvwgy3zao5xxe3debi', { headers: { range: 'bytes=1-2' } })
-      const text = await response.text()
-      return text
-    })
+    expect(statusCode).toBe(206)
+    expect(byteSize).toBe(2)
+    expect(text).toBe('el')
+  })
+
+  test('can get 0-0 byte range from car with missing data', async ({ page, request }) => {
+    const { text, byteSize, statusCode } = await doRangeRequest({ page, range: 'bytes=0-0', path: '/ipfs/QmYhmPjhFjYFyaoiuNzYv8WGavpSRDwdHWe5B4M5du5Rtk' })
+    expect(statusCode).toBe(206)
+    expect(byteSize).toBe(1)
+    expect(text).toBe('+')
+  })
 
-    if (partialText == null) throw new Error('missing response')
+  test('can get trailing byte range from car with missing data', async ({ page }) => {
+    const { bytes, byteSize, statusCode } = await doRangeRequest({ page, range: 'bytes=2200-', path: '/ipfs/QmYhmPjhFjYFyaoiuNzYv8WGavpSRDwdHWe5B4M5du5Rtk' })
+    // TODO: do we need to check the full 872 bytes...?
```

Comment:
idk if theres some block-crossing logic we want to check here, or if checking the size is sufficient, @lidel